# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/cookiebanners/nsCookieBannerTelemetryService.h
# Commit: 517a86043166
# Full Hash: 517a86043166302c58e961b95fc0876d07de6dc6
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2024-03-29 04:51:52
# Regressor Bug: 1874741
# File Overlap Count: 1
# Description:
#   Bug 1874741 - Part 3: Implement the telemetry to collect GDPR choice on Google Search. r=anti-tracking-reviewers,pbz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D205443
# ==============================================================================

diff -r ca7422bd712f -r 517a86043166 toolkit/components/cookiebanners/nsCookieBannerTelemetryService.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/toolkit/components/cookiebanners/nsCookieBannerTelemetryService.h	Thu Mar 28 20:46:34 2024 +0000
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_nsCookieBannerTelemetryService_h__
+#define mozilla_nsCookieBannerTelemetryService_h__
+
+#include "nsICookieBannerTelemetryService.h"
+
+#include "nsIObserver.h"
+
+class nsICookie;
+
+namespace mozilla {
+class nsCookieBannerTelemetryService final
+    : public nsICookieBannerTelemetryService,
+      public nsIObserver {
+  NS_DECL_ISUPPORTS
+  NS_DECL_NSIOBSERVER
+
+  NS_DECL_NSICOOKIEBANNERTELEMETRYSERVICE
+
+ public:
+  static already_AddRefed<nsCookieBannerTelemetryService> GetSingleton();
+
+ private:
+  nsCookieBannerTelemetryService() = default;
+  ~nsCookieBannerTelemetryService() = default;
+
+  bool mIsInitialized = false;
+  bool mIsSearchServiceInitialized = false;
+
+  [[nodiscard]] nsresult Init();
+
+  [[nodiscard]] nsresult Shutdown();
+
+  // Record the telemetry regarding the GDPR choice on Google Search domains.
+  // We only record if the default search engine is Google. When passing no
+  // cookie, we will walk through all cookies under Google Search domains.
+  // Otherwise, we will report the GDPR choice according to the given cookie.
+  [[nodiscard]] nsresult MaybeReportGoogleGDPRChoiceTelemetry(
+      nsICookie* aCookie = nullptr, bool aReportEvent = false);
+};
+
+}  // namespace mozilla
+
+#endif